### ISSUE TITLE
recover mp4.part file

### ISSUE THREAD

**Issue by diga76** [rg3/youtube-dl#10347]
Hello,
sometimes the download don't finish after the end of streaming and I have to break it with ctrl-c.
So I have a mp4.part file and I cannot open it with any media player.
Is there a way to fix this file to have a working mp4 file?
Thank you and congratulations for your great and very usefull software.

**Comment by yan12125** [comment]
Rename xxx.mp4.part to xxx.mp4. If it does not work, run the following command:
```
ffmpeg -i xxx.mp4.part -c copy xxx_fixed.mp4
```

**Comment by diga76** [comment]
I tried the command, but I receive the error:
[mov,mp4,m4a,3gp,3g2,mj2 @ 03289ae0] moov atom not found
xxx.mp4.part: Invalid data found when processing input
Have you other suggestions?
Thanks

**Comment by yan12125** [comment]
How about adding more options to ffmpeg: ```-movflags empty_moov```

**Comment by diga76** [comment]
With the command
ffmpeg -movflags empty_moov -i xxx.mp4.part -c copy xxx_fixed.mp4
I receive the same error.

**Comment by yan12125** [comment]
Which URL are you downloading?

**Comment by diga76** [comment]
It was an old periscope link, now is disappeared.

**Comment by yan12125** [comment]
Sorry, that's all I know. I found lots of "solutions" on Google with "moov atom not found" as the keyword. I didn't try them, though.
For future videos, add ```--hls-prefer-native``` or ```--hls-use-mpegts``` to youtube-dl. With these two options, it's less likely to get broken videos. The downside of these two optoins is that the whole download process is longer and the generated file may not play on all media players.

**Comment by diga76** [comment]
Thank you very much for your support.
This thread will remain closed or is better to set it open ?

**Comment by yan12125** [comment]
I'll keep it closed. There are already some options to prevent incomplete files broken, and recovering a MP4 file is beyond the scope of youtube-dl. This thread can be left for discussions only.

**Comment by diga76** [comment]
Ok, if, in the future, you need to replicate the problem I think it appears, sometimes, on periscope live when the user delete the streaming immediately after the end.
Thank you very much

**Comment by yan12125** [comment]
```--hls-prefer-native``` does not support live streams (yet). Could you try ```---hls-use-mpegts``` in the future?

**Comment by diga76** [comment]
Ok, if I will need future download, I will try with this option.
Thanks

**Comment by Sixem** [comment]
I encountered this issue too and i can confirm that running youtube-dl with `--hls-use-mpegts` and `--prefer-ffmpeg` fixes the issue.

To repair the file after it has been interrupted or whatever you can do `ffmpeg -i "file.mp4.part" -bsf:a aac_adtstoasc -codec copy "file.mp4"`

**Comment by diga76** [comment]
In these days I made some test but the streaming isn't blocked so I can't test if the parameters and the recover procedure works fine. If happen in the future I will test its.
Thank you very much for your suggestions.

**Comment by rezaarifandee** [comment]
I experienced the same problem as you and been able to recover the livestream videos with untrunc. However, you need to have a "similar video" to be used as reference for recovering the moov tags.

sorry for necroing btw.

**Comment by diga76** [comment]
Thank you for you reply, I tried untrunc some months ago, sometimes it works but many times no.

**Comment by ghostis** [comment]
For reference, this version of untrunc works: https://github.com/anthwlock/untrunc

**Comment by diga76** [comment]
Thank you very much for your update

**Comment by akaihola** [comment]
I have a `file1.mp4.part` downloaded from a live stream using `youtube-dl <url>`. It won't play with VLC or convert with `ffmpeg`

I have another `file2.mp4.part` downloaded from a later part of the same live stream using `youtube-dl --hls-use-mpegts`. That one plays fine and `ffmpeg` can process it as well.

I want to view the earlier part in `file1.mp4.part` which is missing from `file2.mp4.part`.

I compiled [untrunc@d72ec32](https://github.com/anthwlock/untrunc/commit/d72ec324fbc29eb00b53c7dafeef09f92308962b) (from 20211-11-21).

I ran `untrunc file2.mp4.part file1.mp4.part` and got:
```
Info: version '' using ffmpeg '4.4.1'
Info: reading ok.mp4
Info: parsing healthy moov atom ... 
Warning: failed decoding atom: invalid atom name: 'B�%'
Error: no 'moov' atom found
```

Is there any way I can recover the video and audio in the first part?
